### ISSUE TITLE
[QST]How to improve performance of read_parquet and url_decode further ?

### ISSUE THREAD

**Issue by chenrui17** [rapidsai/cudf#7571]
**What is your question?**
![image](https://user-images.githubusercontent.com/33319780/110782739-7be74900-82a2-11eb-8808-406280694854.png)
@jlowe As discussed at the meeting yesterday, from this nsight trace, we can see `make_strings_column` cost almost half of the time  , rest time is url_decode and replace .
By the way, my  input parquet  record is long url , length of url about 500~1500.   so this problem is depends on #7545 , right ?

**Comment by jrhemstad** [comment]
I think this is a duplicate of https://github.com/rapidsai/cudf/issues/7545. We know there's room to improve the `make_strings_column` implementation.

**Comment by jlowe** [comment]
Agree mostly this is a duplicate of #7545 but spending 600+ msec on a urldecode isn't ideal either.  We may need to see if there's additional optimizations that can be done there.

@chenrui17 what version (or git commit hash) of cudf is this trace based upon?

**Comment by chenrui17** [comment]
> Agree mostly this is a duplicate of #7545 but spending 600+ msec on a urldecode isn't ideal either. We may need to see if there's additional optimizations that can be done there.
> 
> @chenrui17 what version (or git commit hash) of cudf is this trace based upon?
cudf 0.19  commit id : c76949e5476ce0a190c0f7b1658876846ee7551a

**Comment by kkraus14** [comment]
@jlowe is this addressed now based on the `make_strings_column` optimizations?

**Comment by jlowe** [comment]
`make_strings_column` and `url_decode` are significantly faster on long strings, but we may need some further optimizations for `url_decode` in some cases.  I'm not sure if the trace at the top of this description was captured before or after the recent `url_decode` optimizations.

@chenrui17 please verify whether the new performance of Parquet load and url_decode meets your needs.  If there's additional work needed for just one, I propose tracking that with updated trace/metrics in a new feature request.

**Comment by chenrui17** [comment]
i will propose a new feature request to improve url_decode further
